app/frontend/components/pages/delegations/delegatePage.tsx

```diff
@@ -54,48 +97,100 @@ const Delegate = ({withAccordion, title}: Props): h.JSX.Element => {
   const {
     txSuccessTab,
     stakePool,
+    currentDelegation,
     calculatingDelegationFee,
     delegationFee,
     delegationValidationError,
     gettingPoolInfo,
     isBigDelegator,
     isShelleyCompatible,
-    pool,
     poolRecommendation,
+    validStakepoolDataProvider,
   } = useSelector((state) => ({
+    // REFACTOR: (Untyped errors)
+    delegationValidationError: state.delegationValidationError,
     stakePool: state.shelleyDelegation.selectedPool,
+    currentDelegation: getSourceAccountInfo(state).shelleyAccountInfo.delegation,
     calculatingDelegationFee: state.calculatingDelegationFee,
     delegationFee: state.shelleyDelegation.delegationFee,
-    delegationValidationError: state.delegationValidationError,
     txSuccessTab: state.txSuccessTab,
     gettingPoolInfo: state.gettingPoolInfo,
     isShelleyCompatible: state.isShelleyCompatible,
     poolRecommendation: getSourceAccountInfo(state).poolRecommendation,
-    pool: getSourceAccountInfo(state).shelleyAccountInfo.delegation,
     isBigDelegator: isBigDelegatorSelector(state),
+    validStakepoolDataProvider: state.validStakepoolDataProvider,
   }))
-  const {confirmTransaction} = useActions(actions)
+  const {confirmTransaction, updateStakePoolIdentifier} = useActions(actions)
+
+  // TODO: this reference always changes inside "useEffect" which causes infite loop.
+  // Till this is investigated, this workaround is used instead of "eslint-disable-..."
+  const _updateStakePoolIdentifier = useRef(updateStakePoolIdentifier)
+
+  const [fieldValue, setFieldValue] = useState('')
+  const [error, setError] = useState<Error>(null)
+
+  const handleOnInput = (event: any): void => {
+    const fieldValue: string = event?.target?.value
+    setFieldValue(fieldValue)
+
+    // Note: this could be debounced
+    if (fieldValue) {
+      const {poolHash, error} = validateInput(fieldValue, validStakepoolDataProvider)
+      if (!error) {
+        /*
+        Relates to REFACTOR (calculateFee):
```

Comment:
I think that it would be ideal if this component just calculated plan, rendered modal, pass data, ... and no need for global state.